### PR TITLE
feature/APPS-479-disable-unsupported-systems-in-systems-list

### PR DESCRIPTION
**Pull request by AdamBraun**

### CODE REVIEW

File: src/providers/AppContext/reducer.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ export const initialState = {
   environments: {} as WalkMeEnvironment[],
   parsedEnvironments: [] as IWMDropdownOption[],
   dateRange: defaultDateRange,
-  systems: [] as SystemData[],
```
Comment (yuvallevanoni) [inline src/providers/AppContext/reducer.ts:29]: In line 25 there's the single 'system' obj.
It's of type 'SystemData'
Should it also change to type 'TeachMeSystemData'?

**Comment by yuvallevanoni** [comment]
I only commented one question

File: src/providers/AppContext/reducer.ts (hunk 1 above)
Comment (AdamBraun) [reply src/providers/AppContext/reducer.ts:29]: No. Or, at least, not now.
Right now I only wanted to filter out what systems are enabled.
If we do find the place where we want to know if the current system is a disabled one, we can easily add it

Comment (yuvallevanoni) [reply src/providers/AppContext/reducer.ts:29]: Great!
how do I approve this PR???

**Review by yuvallevanoni — APPROVED** [review]

File: src/providers/AppContext/app-context.interface.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export interface IAction {
   user?: UserData;
   originalUser?: UserData;
   system?: SystemData | string; // string type for non-system-users - temporary solution
```
Comment (yuvallevanoni) [inline src/providers/AppContext/app-context.interface.ts:21]: remove comment

File: src/providers/AppContext/app-context.interface.ts [hunk 3]
```diff
@@ -38,6 +39,7 @@ export interface IState {
   user: UserData;
   originalUser: UserData;
   system: SystemData | string; // string type for non-system-users - temporary solution
```
Comment (yuvallevanoni) [inline src/providers/AppContext/app-context.interface.ts:41]: remove comment

File: src/components/Layout/HeaderToolbar/SystemMenu.tsx [hunk 4]
```diff
@@ -21,6 +21,13 @@ export default function TeachMeSystemData({ className }: { className?: string })
   const systemIsValid = useMemo(() => system !== '' && (system as SystemData)?.userId, [system]);
 
   const handleMenuClick = (selected: IWMDropdownOption) => {
+    const parsedSystem = Array.isArray(parsedSystems)
```
Comment (yuvallevanoni) [inline src/components/Layout/HeaderToolbar/SystemMenu.tsx:24]: Is it considered ok that there are two 'parsedSystem' variables? (one is inside the find method callback)
*I always thought it's better to give different names, just to prevent confusion (of course it doesn't change the functionality)

**Review by yuvallevanoni — APPROVED** [review]

File: src/components/Layout/HeaderToolbar/SystemMenu.tsx (hunk 4 above)
Comment (chen-c-walkme) [reply src/components/Layout/HeaderToolbar/SystemMenu.tsx:24]: I agree. It doesn't matter if it works or not, we should avoid double naming anyway, so it will be easier to read.

File: src/providers/AppContext/app-context.interface.ts (hunk 3 above)
Comment (AdamBraun) [reply src/providers/AppContext/app-context.interface.ts:41]: Not my code. I have no clue why it's a temporary solution or whether we need it

File: src/components/Layout/HeaderToolbar/SystemMenu.tsx [hunk 5]
```diff
@@ -11,13 +11,23 @@ import WMButton, { IconPosition } from '../../common/WMButton';
 
 import { parseSystems } from './utils';
 
-export default function SystemMenu({ className }: { className?: string }): ReactElement {
+export interface TeachMeSystemData extends SystemData {
+  disabled: boolean;
+}
+export default function TeachMeSystemData({ className }: { className?: string }): ReactElement {
```
Comment (chen-c-walkme) [inline src/components/Layout/HeaderToolbar/SystemMenu.tsx:17]: Why did you change the name from 'SystemMenu' to 'TeachMeSystemData'? Notice this is the same name as the interface you added..

File: src/components/Layout/HeaderToolbar/SystemMenu.tsx (hunk 4 above)
Comment (AdamBraun) [reply src/components/Layout/HeaderToolbar/SystemMenu.tsx:24]: It's called `shadowing`.
Usually a very bad practice, because of all the confusion and human errors it can create.
Here' specifically, it's not shadowing as the two variables do not exist at the same time.
I also double checked by locally adding the `no-shadow` rule in `eslintrc.js` and running it on the file. It did not report a warning or an error regarding this.

Personally, I feel it makes sense, because I'm finding a specific parsed system and saving it in a variable.

Still, if you feel it's confusing, I can change it

File: src/components/Layout/HeaderToolbar/SystemMenu.tsx (hunk 5 above)
Comment (AdamBraun) [reply src/components/Layout/HeaderToolbar/SystemMenu.tsx:17]: O_O Good catch. I have no clue...
I guess I missed the mistake because functionally it doesn't matter, so the compiler didn't care

File: src/components/Layout/HeaderToolbar/SystemMenu.tsx (hunk 4 above)
Comment (chen-c-walkme) [reply src/components/Layout/HeaderToolbar/SystemMenu.tsx:24]: I get what you're saying, but still, in order to avoid confusion and chances of human error, I'd rather name it differently.
